weave/artifact_wandb.py

```diff
@@ -462,7 +462,12 @@ def add_artifact_reference(self, uri: str) -> None:
         # but the SDK add_reference call requires something like:
         # wandb-artifact://41727469666163743a353432353830303535/obj.object.json
         assert uri.startswith("wandb-artifact:///")
-        uri_path = "/".join(uri.split("/")[3:-1])  # Trim off protocol and /obj
+        uri_parts = uri.split("/")
```

Comment:
nit: perhaps consider using `WeaveWBArtifactURI.parse` then using the parsed fields.